src/github.com/getlantern/flashlight/logging/logging_test.go

```diff
@@ -7,12 +7,36 @@ import (
 	"regexp"
 	"strings"
 	"testing"
+	"time"
 
+	"github.com/getlantern/eventual"
 	"github.com/getlantern/go-loggly"
 	"github.com/getlantern/golog"
 	"github.com/stretchr/testify/assert"
 )
 
+// Test to make sure user agent registration, listening, etc is all working.
+func TestUserAgent(t *testing.T) {
+	agent := "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.86 Safari/537.36"
+	userAgent := eventual.NewValue()
+
+	listener := func(a string) {
+		userAgent.Set(a)
+	}
+	AddUserAgentListener(listener)
+	RegisterUserAgent(agent)
+
+	// Just register again to hit the code path for duplicates.
+	RegisterUserAgent(agent)
```

Comment:
In order to test eventual assignation (the purpose of eventual), perhaps we can run `RegisterUserAgent` in a different goroutine that waits for couple hundred miliseconds to register the agent, so `received` needs to block for that time.
These can be two separate tests though. One with immediately available value and one in which eventual blocks.